canonical_mutation.cc

```diff
@@ -89,3 +90,81 @@ mutation canonical_mutation::to_mutation(schema_ptr s) const {
     }
     return m;
 }
+
+static sstring bytes_to_text(bytes_view bv) {
+    sstring ret(sstring::initialized_later(), bv.size());
+    std::copy_n(reinterpret_cast<const char*>(bv.data()), bv.size(), ret.data());
+    return ret;
+}
+
+std::ostream& operator<<(std::ostream& os, const canonical_mutation& cm) {
+    auto in = ser::as_input_stream(cm._data);
+    auto mv = ser::deserialize(in, boost::type<ser::canonical_mutation_view>());
+    column_mapping mapping = mv.mapping();
+    auto partition_view = mutation_partition_view::from_view(mv.partition());
+    fmt::print(os, "{{canonical_mutation: ");
+    fmt::print(os, "table_id {} schema_version {} ", mv.table_id(), mv.schema_version());
+    fmt::print(os, "partition_key {} ", mv.key());
```

Comment:
There is a pretty printer for partition keys as well. Wouldn't it be better to use that instead of the raw hex representation?

Reply:
Again I can't use it because the schema isn't available.

We could add a pretty printer that makes use of the/a schema, but that's against the spirit of canonical_mutation which is supposed to be portable.